src/LogoFX.Client.Core.Platform/src/PlatformDispatch.cs

```diff
@@ -24,19 +36,34 @@ private void EnsureDispatch()
         /// </summary>
         public void InitializeDispatch()
         {
-            var dispatcher = Dispatcher.CurrentDispatcher;
+	        var dispatcher
+#if !WINUI3
+                = Dispatcher.CurrentDispatcher;
+#else
+		        = DispatcherQueue.GetForCurrentThread();
+#endif
             if (dispatcher == null)
                 throw new InvalidOperationException("Dispatch is not initialized correctly");
-            _dispatch = (action, @async, priority) =>
+            _dispatch = (action, async, priority) =>
             {
-                if (!@async && dispatcher.CheckAccess())
+                if (!async &&
```

Comment:
@glyad In short swap all similar conditions

Reply:
Fixed